pkg/apis/servicecatalog/types.go

```diff
@@ -443,3 +457,19 @@ const (
 const (
 	FinalizerServiceCatalog string = "kubernetes-incubator/service-catalog"
 )
+
+// ParametersFromSource represents the source of a set of Parameters
+type ParametersFromSource struct {
```

Comment:
We have indirection to allow for future methods in a forward compatible manner and is conventional for how we solve the problem of differentiating different options for behaviors in kubernetes APIs.

> ParametersFrom []SecretKeyRef

This is not forward compatible with other modes of addressing parameter payloads.  Say for example that you want to add a similar behavior that uses configmaps instead of secrets.  With @nilebox's proposal, you add a new field to ParametersFromSource.  With your proposal, you have to add a whole new field.

Does that make sense?